BuddiesTests/Utilities/AlgoliaTest.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  AlgoliaTest.swift
+//  BuddiesTests
+//
+//  Created by Noah Allen on 2/15/19.
+//  Copyright © 2019 Jack and the Beans. All rights reserved.
+//
+
+import Foundation
+import InstantSearchClient
+
+import XCTest
+@testable import Buddies
+
+class AlgoliaTest: XCTestCase {
+    func testSearchWithText() {
+        let client = TestClient(appID: "foo", apiKey: "bar")
+        let search = AlgoliaSearch(algoliaClient: client)
+        let index = TestIndex()
+        let textQuery = "Mickey Mouse"
+        search.searchActivities(withText: textQuery, usingIndex: index) { (ids, err) in
+            var checker = false
+            XCTAssert(ids.count > 0, "The query completed.")
+            if let query = index.query {
+                checker = query.query == textQuery
+            }
+            XCTAssert(checker, "Constructs textual query correctly.")
+        }
+    }
+    func testSearchWithTopics() {
+        let client = TestClient(appID: "foo", apiKey: "bar")
+        let search = AlgoliaSearch(algoliaClient: client)
+        let index = TestIndex()
+        let topics = ["a", "b"]
+        search.searchActivities(matchingAnyTopicOf: topics, usingIndex: index) { (ids, err) in
+            var checker = false
+            XCTAssert(ids.count > 0, "The query completed.")
+            if let query = index.query {
+                let filterText = query.filters
+                checker = filterText == "(topic_ids:a OR topic_ids:b)"
+            }
+            XCTAssert(checker, "Constructs topic query correctly.")
+        }
+    }
+    func testSearchWithDateAndTopics() {
+        let client = TestClient(appID: "foo", apiKey: "bar")
+        let search = AlgoliaSearch(algoliaClient: client)
+        let index = TestIndex()
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy/MM/dd"
+        let start = formatter.date(from: "2019/02/15")
+        let end = formatter.date(from: "2019/02/20")
+        let topics = ["a", "b"]
+        search.searchActivities(matchingAnyTopicOf: topics, startingAt: start, endingAt: end, usingIndex: index) { (ids, err) in
+            var checker = false
+            XCTAssert(ids.count > 0, "The query completed.")
+            if let query = index.query {
+                let filterText = query.filters
+                checker = filterText == "(end_time_num >= 1550206800000 AND start_time_num <= 1550638800000) AND (topic_ids:a OR topic_ids:b)"
+            }
+            XCTAssert(checker, "Constructs topic AND date query correctly.")
```

Comment:
For my date debugging stuff, I had to bite the bullet and print the output to see what was going on in Travis.

Reply:
Doing that now.